### ISSUE TITLE
CI with Java 8

### ISSUE THREAD

**Issue by mccartney** [knowm/XChart#757]
- The project claims compatibility with Java 8.
- Yet our only CI in PRs executes against Java 11:
https://github.com/knowm/XChart/blob/develop/.github/workflows/maven_on_pull_request.yml#L20
Thus we risk bringing in something which won't work with Java 8.

I suggest to add CI check using Java 8.

**Comment by mccartney** [comment]
The good thing is that current [`develop`](7994017fa4283cde1294fbaf864910cd5ae401a4) seems to pass with Java 8 when tested locally.

**Comment by mccartney** [comment]
From https://github.com/spotify/fmt-maven-plugin#using-with-java-8
> Using with Java 8
> Starting from version 1.8, Google Java Formatter requires Java 11 to run. Incidently, all versions of this plugin starting from 2.10 inclusively also require this Java version to properly function. The 2.9.x release branch is the most up-to-date version that still runs on Java 8.

**Comment by timmolter** [comment]
yes, we should add a CI check using Java 8. Also yes, I recall the constraint on the formatter. Maybe we just leave it at 2.9.x.

**Comment by mccartney** [comment]
Yeah. Downgrading to `2.9.1` of the format plugin adds a lot of changes like
![image](https://github.com/knowm/XChart/assets/762437/2ddb9424-cf7a-4bc3-8854-ea351f21ee5a)
(99 files)

**Comment by mccartney** [comment]
Currently I am thinking about:
- sticking with Java 8 as the library dependency
- using the latest of the format plugin
- making the CI run in two modes: Java 11 (as is) and Java 8 (which would use Java 11 to build and just Java 8 to run tests).

**Comment by timmolter** [comment]
The main point of the formatter is to have a standard where all committers use so that merge diffs are best readable. That's really the only reason. We can remove all the author comments as well.

**Comment by timmolter** [comment]
Closing as obsolete — the project has since moved its minimum Java target to Java 11, so the discrepancy between the claimed Java 8 compatibility and the CI running on Java 11 no longer exists. No CI changes needed.
